apps/api/src/app/controllers/__tests__/createTestModule.ts

```diff
@@ -3,24 +3,24 @@ import { Test } from '@nestjs/testing';
 import { ArangoConnectionProvider } from '../../../persistence/database/arango-connection.provider';
 import { DatabaseProvider } from '../../../persistence/database/database.provider';
 import { RepositoryProvider } from '../../../persistence/repositories/repository.provider';
+import { DTO } from '../../../types/DTO';
 import buildConfigFilePath from '../../config/buildConfigFilePath';
 import { Environment } from '../../config/constants/Environment';
+import { EnvironmentVariables } from '../../config/env.validation';
 import buildMockConfigServiceSpec from '../../config/__tests__/utilities/buildMockConfigService';
 import { CategoryController } from '../category.controller';
 import { EdgeConnectionController } from '../edgeConnection.controller';
 import { ResourceViewModelController } from '../resourceViewModel.controller';
 import { TagController } from '../tag.controller';
 
-export default async (testDatabaseName: string) =>
+export default async (configOverrides: Partial<DTO<EnvironmentVariables>>) =>
     Test.createTestingModule({
         providers: [
             {
                 provide: ConfigService,
                 useFactory: () =>
```

Comment:
Does useFactory create classes dynamically inside a module?

Reply:
You're close. The module is what is being dynamically created. A factory creates instances of a class, not a new class (that would be a mixin or decorator). The simplest way to build an instance of a class is to call the constructor directly. Sometimes that's not enough. In our repositories layer, the factory calls an invariant validator on the incoming DTO first before newing up the instance, for example. 

Each Nest (same as Angular) module has a set of providers that can provide dependencies for dependency injection. For example, when one of the classes in this module needs a private reference to a parameter of type `ArangoConnectionProvider` in its constructor, nest will inject one for us, using the factory we specify in `useFactory`. It's key that this can be async. That allows us to await initializing the database before moving on. In fact, that's why our `PersistenceModule` in the actual live code path is a dynamic module.